config/Dockerfiles/JenkinsfileContainerBuild

```diff
@@ -0,0 +1,209 @@
+//            hostPathVolume(mountPath: '/var/lib/containers', hostPath: '/var/lib/containers')
+            emptyDirVolume(mountPath: '/var/run/containers', memory: false),
+            emptyDirVolume(mountPath: '/var/lib/containers', memory: false)
+        ]
+)
+
+{
+    node(podName) {
+        ansiColor('xterm') {
+            timestamps {
+                // We need to set env.HOME because the openshift slave image
+                // forces this to /home/jenkins and then ~ expands to that
+                // even though id == "root"
+                // See https://github.com/openshift/jenkins/blob/master/slave-base/Dockerfile#L5
+                //
+                // Even the kubernetes plugin will create a pod with containers
+                // whose $HOME env var will be its workingDir
+                // See https://github.com/jenkinsci/kubernetes-plugin/blob/master/src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java#L311
+                //
+                env.HOME = "/root"
+                //
+                try {
+                    // Prepare our environment
+                    stage("Git Checkout LinchPin") {
+                        deleteDir()
+                        // Set our default env variables. Many of these
+                        // are ci-pipeline specific, but we do need some
+                        pipelineUtils.setDefaultEnvVars()
+                        // Decorate our build
+                        currentBuild.displayName = " LinchPin Container Build#: ${env.BUILD_NUMBER}"
+
+                        // Gather some info about the node we are running on for diagnostics
+                        pipelineUtils.verifyPod(OPENSHIFT_NAMESPACE, env.NODE_NAME)
+                        dir('linchpin') {
+                            // Checkout our linchpin repo based on the value of env.ghprbActualCommit
+                            checkout([$class: 'GitSCM',
+                                      branches: [[name: env.ghprbActualCommit]],
+                                      doGenerateSubmoduleConfigurations: false,
+                                      extensions                       : [],
+                                      submoduleCfg                     : [],
+                                      userRemoteConfigs                : [
+                                              [refspec:
+                                                       '+refs/heads/*:refs/remotes/origin/*  +refs/pull/*:refs/remotes/origin/pr/* ',
+                                               url: "https://github.com/${env.ghprbGhRepository}"]
+                                      ]
+                            ])
+                            def creds = com.cloudbees.plugins.credentials.CredentialsProvider.lookupCredentials(
+                              com.cloudbees.plugins.credentials.Credentials.class
```

Comment:
Is creds used for anything here?

Reply:
For pushing up to the docker hub? I'm not sure if this is the credentials store in Jenkins, but that's what I believe it is fore.

Reply:
I don't think creds is used at all.  You have another withCredentials down below which does look like it's used for pushing to docker.